jgrapht-core/src/test/java/org/jgrapht/graph/AsSynchronizedGraphTest.java

```diff
@@ -0,0 +1,500 @@
+/*
+ * (C) Copyright 2003-2018, by CHEN Kui and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.graph;
+
+import java.util.*;
+import junit.extensions.*;
+import junit.framework.*;
+import junit.textui.*;
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+/**
+ * Test class AsSynchronizedGraph.
+ *
+ * @author CHEN Kui
+ */
+public class AsSynchronizedGraphTest
+{
+    private Vector<Integer> vertices;
+    private Vector<DefaultEdge> edges;
```

Comment:
I question the use of `Vector`s here. AFAICT, they're not being used in a concurrent context, which I believe is what `Vector`s specifically are (poorly) designed for, so I'd strongly encourage to swap them out for `List`s.

Reply:
Actually this is one of the rare cases where Vector is OK; see the access in TestThread (which explicitly synchronizes as well).

Reply:
Oh, cool, I'd missed those usages. I'm not sure I understand yet why `Vector` is OK here, but I admit I've not looked at the code deeply, so I'd be perfectly happy to leave things as-is in this case. :)

Reply:
Actually, wait...I don't see where vertex(), addEdge(), etc are called.  @Yimismi is that dead code from a previous iteration?  If so, I agree that the Vectors should be changed to ArrayList.

Reply:
In fact, both ArrayList and Vector are OK. I will change it.